components/client/src/main/java/com/hotels/styx/client/StyxHttpClient.java

```diff
@@ -336,17 +333,24 @@ public String toString() {
      * A builder for {@link com.hotels.styx.client.StyxHttpClient}.
      */
     public static class Builder {
-        private final BackendService backendService;
+
+        private final Id backendServiceId;
         private MetricRegistry metricsRegistry = new CodaHaleMetricRegistry();
         private List<RewriteRule> rewriteRules = emptyList();
         private RetryPolicy retryPolicy = new RetryNTimes(3);
         private LoadBalancer loadBalancer;
         private boolean contentValidation;
         private OriginStatsFactory originStatsFactory;
         private String originsRestrictionCookieName;
+        private StickySessionConfig stickySessionConfig = stickySessionDisabled();
 
-        public Builder(BackendService backendService) {
-            this.backendService = checkNotNull(backendService);
+        public Builder(Id backendServiceId) {
+            this.backendServiceId = requireNonNull(backendServiceId);
+        }
+
+        public Builder stickySessionConfig(StickySessionConfig stickySessionConfig) {
```

Comment:
Sticky session config is not called from the client factory.

That needs to be added in, and ideally testing hardened to catch this issue.